migrator/replica/db_replica_manager_impl.go

```diff
@@ -0,0 +1,117 @@
+package replica
+
+import (
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/stackrox/rox/migrator/replica/postgres"
+	"github.com/stackrox/rox/migrator/replica/rocksdb"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/migrations"
+)
+
+// DBReplicaManagerImpl - scans and manage database replicas within central.
+type DBReplicaManagerImpl struct {
+	forceRollbackVersion string
+	adminConfig          *pgxpool.Config
+	sourceMap            map[string]string
+	basePath             string
+	dbmRocks             *rocksdb.DBReplicaManagerImpl
+	dbmPostgres          *postgres.DBReplicaManagerImpl
+}
+
+// NewPostgres - returns a new ready-to-use store.
+func NewPostgres(basePath string, forceVersion string, adminConfig *pgxpool.Config, sourceMap map[string]string) *DBReplicaManagerImpl {
+	return &DBReplicaManagerImpl{
+		forceRollbackVersion: forceVersion,
+		adminConfig:          adminConfig,
+		sourceMap:            sourceMap,
+		basePath:             basePath,
+		dbmRocks:             rocksdb.New(basePath, forceVersion),
+		dbmPostgres:          postgres.New(forceVersion, adminConfig, sourceMap),
+	}
+}
+
+// New - returns a new ready-to-use store.
+func New(basePath string, forceVersion string) *DBReplicaManagerImpl {
+	return &DBReplicaManagerImpl{
+		forceRollbackVersion: forceVersion,
+		basePath:             basePath,
+		dbmRocks:             rocksdb.New(basePath, forceVersion),
+	}
+}
+
+// Scan - checks the persistent data of central and gather the replica information
+// from the supported databases.
+func (d *DBReplicaManagerImpl) Scan() error {
+	err := d.dbmRocks.Scan()
+	if err != nil {
+		// If our focus is Postgres, just log the error and ignore Rocks
+		if features.PostgresDatastore.Enabled() {
+			log.Warn(err)
+		} else {
+			return err
+		}
+	}
+
+	if features.PostgresDatastore.Enabled() {
+		err = d.dbmPostgres.Scan()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// GetReplicaToMigrate - finds a replica to migrate.
+// It returns the replica link, path to database, postgres database name and error if fails.
+func (d *DBReplicaManagerImpl) GetReplicaToMigrate() (string, string, string, error) {
+	// Get the RocksDB replica we are migrating
+	replica, replicaPath, err := d.dbmRocks.GetReplicaToMigrate()
+	if err != nil {
+		return "", "", "", err
```

Comment:
Why are we return no replica chosen here? We did not try to select a replica from Postgres yet.

Reply:
nice catch as that could occur if the PVC were removed.  Will fix.